File: app/src/main/java/org/rm3l/ddwrt/resources/SpeedTestResult.java
package org.rm3l.ddwrt.resources;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import org.rm3l.ddwrt.actions.activity.PingRTT;

/**
 * Created by rm3l on 12/01/16.
 */
public class SpeedTestResult {

    /**
     * the internal id (in DB)
     */
    private long id = -1l;

    @NonNull
    private String router;

    @NonNull
    //YYYY-MM-dd
    private String date;

    @NonNull
    private Number wanPing;

    @NonNull
    private Number wanDl;

    @NonNull
    private Number wanUl;

    /**
     * Useless ?
     */
    @Nullable
    private String connectionType;

    @Nullable
    private Number connectionDl;

    @Nullable
    private Number connectionUl;

    @NonNull
    private String server;

    @Nullable
    private String serverCountryCode;

    private PingRTT wanPingRTT;

    private Number wanDLFileSize;

    private Long wanDLDuration;

    private Number wanULFileSize;

    private Long wanULDuration;

    private Number connectionDLFileSize;

    private Long connectionDLDuration;

    private Number connectionULFileSize;

    private Long connectionULDuration;

    public SpeedTestResult(@NonNull String router,
                           @NonNull String date,
                           @NonNull String server,
                           @NonNull Number wanPing,
                           @NonNull Number wanDl,
                           @NonNull Number wanUl,
                           @Nullable String connectionType,
                           @Nullable Number connectionDl,
                           @Nullable Number connectionUl,
                           @Nullable String serverCountryCode) {

        this.router = router;
        this.date = date;
        this.wanDl = wanDl;
        this.wanUl = wanUl;
        this.connectionType = connectionType;
        this.connectionDl = connectionDl;
        this.connectionUl = connectionUl;
        this.server = server;
        this.serverCountryCode = serverCountryCode;
        this.wanPing = wanPing;
        this.wanPingRTT = new PingRTT()
                .setAvg(wanPing.floatValue());
    }

    public SpeedTestResult() {}

    public SpeedTestResult setRouter(@NonNull String router) {
        this.router = router;
        return this;
    }

    public SpeedTestResult setDate(@NonNull String date) {
        this.date = date;
        return this;
    }

    public SpeedTestResult setWanPing(@NonNull Number wanPing) {
        this.wanPing = wanPing;
        return this;
    }

    public SpeedTestResult setWanDl(@NonNull Number wanDl) {
        this.wanDl = wanDl;
        return this;
    }

    public SpeedTestResult setWanUl(@NonNull Number wanUl) {
        this.wanUl = wanUl;
        return this;
    }

    public SpeedTestResult setConnectionType(@Nullable String connectionType) {
        this.connectionType = connectionType;
        return this;
    }

    public SpeedTestResult setConnectionDl(@Nullable Number connectionDl) {
        this.connectionDl = connectionDl;
        return this;
    }

    public SpeedTestResult setConnectionUl(@Nullable Number connectionUl) {
        this.connectionUl = connectionUl;
        return this;
    }

    public SpeedTestResult setServer(@NonNull String server) {
        this.server = server;
        return this;
    }

    public SpeedTestResult setServerCountryCode(@Nullable String serverCountryCode) {
        this.serverCountryCode = serverCountryCode;
        return this;
    }

    public long getId() {
        return id;
    }

    public SpeedTestResult setId(long id) {
        this.id = id;
        return this;
    }

    @Nullable
    public String getServerCountryCode() {
        return serverCountryCode;
    }

    @NonNull
    public String getRouter() {
        return router;
    }

    @NonNull
    public String getDate() {
        return date;
    }

    @NonNull
    public Number getWanPing() {
        return wanPing;
    }

    @NonNull
    public Number getWanDl() {
        return wanDl;
    }

    @NonNull
    public Number getWanUl() {
        return wanUl;
    }

    @Nullable
    public String getConnectionType() {
        return connectionType;
    }

    @Nullable
    public Number getConnectionDl() {
        return connectionDl;
    }

    @Nullable
    public Number getConnectionUl() {
        return connectionUl;
    }

    @NonNull
    public String getServer() {
        return server;
    }

    public PingRTT getWanPingRTT() {
        return wanPingRTT;
    }

    public SpeedTestResult setWanPingRTT(PingRTT wanPingRTT) {
        this.wanPingRTT = wanPingRTT;
        if (wanPingRTT != null) {
            this.setWanPing(wanPingRTT.getAvg());
        }
        return this;
    }

    public Number getWanDLFileSize() {
        return wanDLFileSize;
    }

    public SpeedTestResult setWanDLFileSize(Number wanDLFileSize) {
        this.wanDLFileSize = wanDLFileSize;
        return this;
    }

    public Long getWanDLDuration() {
        return wanDLDuration;
    }

    public SpeedTestResult setWanDLDuration(Long wanDLDuration) {
        this.wanDLDuration = wanDLDuration;
        return this;
    }

    public Number getWanULFileSize() {
        return wanULFileSize;
    }

    public SpeedTestResult setWanULFileSize(Number wanULFileSize) {
        this.wanULFileSize = wanULFileSize;
        return this;
    }

    public Long getWanULDuration() {
        return wanULDuration;
    }

    public SpeedTestResult setWanULDuration(Long wanULDuration) {
        this.wanULDuration = wanULDuration;
        return this;
    }

    public Number getConnectionDLFileSize() {
        return connectionDLFileSize;
    }

    public SpeedTestResult setConnectionDLFileSize(Number connectionDLFileSize) {
        this.connectionDLFileSize = connectionDLFileSize;
        return this;
    }

    public Long getConnectionDLDuration() {
        return connectionDLDuration;
    }

    public SpeedTestResult setConnectionDLDuration(Long connectionDLDuration) {
        this.connectionDLDuration = connectionDLDuration;
        return this;
    }

    public Number getConnectionULFileSize() {
        return connectionULFileSize;
    }

    public SpeedTestResult setConnectionULFileSize(Number connectionULFileSize) {
        this.connectionULFileSize = connectionULFileSize;
        return this;
    }

    public Long getConnectionULDuration() {
        return connectionULDuration;
    }

    public SpeedTestResult setConnectionULDuration(Long connectionULDuration) {
        this.connectionULDuration = connectionULDuration;
        return this;
    }
}

File: app/src/main/java/org/rm3l/router_companion/resources/SpeedTestResult.kt
package org.rm3l.router_companion.resources

import org.rm3l.router_companion.actions.activity.PingRTT

/**
 * Created by rm3l on 12/01/16.
 */
class SpeedTestResult {

    /**
     * the internal id (in DB)
     */
    private var id = -1L

    private var router: String? = null

    private //YYYY-MM-dd
    var date: String? = null

    private var wanPing: Number? = null

    private var wanDl: Number? = null

    private var wanUl: Number? = null

    /**
     * Useless ?
     */
    private var connectionType: String? = null

    private var connectionDl: Number? = null

    private var connectionUl: Number? = null

    private var server: String? = null

    private var serverCountryCode: String? = null

    private var wanPingRTT: PingRTT? = null

    private var wanDLFileSize: Number? = null

    private var wanDLDuration: Long? = null

    private var wanULFileSize: Number? = null

    private var wanULDuration: Long? = null

    private var connectionDLFileSize: Number? = null

    private var connectionDLDuration: Long? = null

    private var connectionULFileSize: Number? = null

    private var connectionULDuration: Long? = null

    constructor(router: String, date: String, server: String,
                wanPing: Number, wanDl: Number, wanUl: Number,
                connectionType: String?, connectionDl: Number?, connectionUl: Number?,
                serverCountryCode: String?) {

        this.router = router
        this.date = date
        this.wanDl = wanDl
        this.wanUl = wanUl
        this.connectionType = connectionType
        this.connectionDl = connectionDl
        this.connectionUl = connectionUl
        this.server = server
        this.serverCountryCode = serverCountryCode
        this.wanPing = wanPing
        this.wanPingRTT = PingRTT().setAvg(wanPing.toFloat())
    }

    constructor()

    fun getId(): Long {
        return id
    }

    fun setId(id: Long): SpeedTestResult {
        this.id = id
        return this
    }

    fun getServerCountryCode(): String? {
        return serverCountryCode
    }

    fun setServerCountryCode(serverCountryCode: String?): SpeedTestResult {
        this.serverCountryCode = serverCountryCode
        return this
    }

    fun getRouter(): String? {
        return router
    }

    fun setRouter(router: String?): SpeedTestResult {
        this.router = router
        return this
    }

    fun getDate(): String? {
        return date
    }

    fun setDate(date: String?): SpeedTestResult {
        this.date = date
        return this
    }

    fun getWanPing(): Number? {
        return wanPing
    }

    fun setWanPing(wanPing: Number?): SpeedTestResult {
        this.wanPing = wanPing
        return this
    }

    fun getWanDl(): Number? {
        return wanDl
    }

    fun setWanDl(wanDl: Number?): SpeedTestResult {
        this.wanDl = wanDl
        return this
    }

    fun getWanUl(): Number? {
        return wanUl
    }

    fun setWanUl(wanUl: Number?): SpeedTestResult {
        this.wanUl = wanUl
        return this
    }

    fun getConnectionType(): String? {
        return connectionType
    }

    fun setConnectionType(connectionType: String?): SpeedTestResult {
        this.connectionType = connectionType
        return this
    }

    fun getConnectionDl(): Number? {
        return connectionDl
    }

    fun setConnectionDl(connectionDl: Number?): SpeedTestResult {
        this.connectionDl = connectionDl
        return this
    }

    fun getConnectionUl(): Number? {
        return connectionUl
    }

    fun setConnectionUl(connectionUl: Number?): SpeedTestResult {
        this.connectionUl = connectionUl
        return this
    }

    fun getServer(): String? {
        return server
    }

    fun setServer(server: String?): SpeedTestResult {
        this.server = server
        return this
    }

    fun getWanPingRTT(): PingRTT? {
        return wanPingRTT
    }

    fun setWanPingRTT(wanPingRTT: PingRTT?): SpeedTestResult {
        this.wanPingRTT = wanPingRTT
        if (wanPingRTT != null) {
            this.setWanPing(wanPingRTT.avg)
        }
        return this
    }

    fun getWanDLFileSize(): Number? {
        return wanDLFileSize
    }

    fun setWanDLFileSize(wanDLFileSize: Number?): SpeedTestResult {
        this.wanDLFileSize = wanDLFileSize
        return this
    }

    fun getWanDLDuration(): Long? {
        return wanDLDuration
    }

    fun setWanDLDuration(wanDLDuration: Long?): SpeedTestResult {
        this.wanDLDuration = wanDLDuration
        return this
    }

    fun getWanULFileSize(): Number? {
        return wanULFileSize
    }

    fun setWanULFileSize(wanULFileSize: Number?): SpeedTestResult {
        this.wanULFileSize = wanULFileSize
        return this
    }

    fun getWanULDuration(): Long? {
        return wanULDuration
    }

    fun setWanULDuration(wanULDuration: Long?): SpeedTestResult {
        this.wanULDuration = wanULDuration
        return this
    }

    fun getConnectionDLFileSize(): Number? {
        return connectionDLFileSize
    }

    fun setConnectionDLFileSize(connectionDLFileSize: Number?): SpeedTestResult {
        this.connectionDLFileSize = connectionDLFileSize
        return this
    }

    fun getConnectionDLDuration(): Long? {
        return connectionDLDuration
    }

    fun setConnectionDLDuration(connectionDLDuration: Long?): SpeedTestResult {
        this.connectionDLDuration = connectionDLDuration
        return this
    }

    fun getConnectionULFileSize(): Number? {
        return connectionULFileSize
    }

    fun setConnectionULFileSize(connectionULFileSize: Number?): SpeedTestResult {
        this.connectionULFileSize = connectionULFileSize
        return this
    }

    fun getConnectionULDuration(): Long? {
        return connectionULDuration
    }

    fun setConnectionULDuration(connectionULDuration: Long?): SpeedTestResult {
        this.connectionULDuration = connectionULDuration
        return this
    }
}

